developer-experience/recipes/onboarding-guide-template.md

```diff
@@ -0,0 +1,35 @@
+# Onboarding Guide Template
+
+When developing an onboarding document for a team, it should contain details of engagement scope, team processes, codebase, coding standards, team agreements, software requirements and setup details.
+
+## Overview and Goals
+
+* List a few sentences explaining the high-level summary and the scope of the engagement.
+* Consider adding any additional background and context as needed.
+* Include the goals of this engagement and what the team is trying to achieve.
+
+## Contacts
+
+* List a few of the main contacts from the engagement both from the customer and Microsoft team.
+* Consider including the roles of these main contacts so that the team knows who to reach out to depending on the situation.
+
+## Team Agreement and Code of Conduct
+
+* Include the team's code of conduct or agreement that defines a set of expectation from each team member and how the team has agreed to operate.
+* This could include Pull request process, coding style, coding standards, definition of done.
+* List a few bullets on communication expectations, meeting timings.
+
+## Dev Environment Setup
+
+* List out steps to run the project end-to-end.
```

Comment:
again would try to steer people towards maybe using this page as an index to other content if it already exists elsewhere. seems common sense but, dont like to assume :)

Reply:
For this specific one - a template for the dev env setup does not currently exist in the repo and I wasn't able to find one that i can include here as an example as each project would be different in terms of the setup. I can connect with you more on this and see what would be the best way for this section.

Reply:
I think @charleszipp was saying that the `on-boarding guide` document might be a good place to have an index of project specific documents. Not that this `.md` file needs to be that link.

Just my understanding :)

Reply:
Updated the description to be more clear on this so that this guide can be kept as concise and effective as possible.